public/app/plugins/datasource/prometheus/types.ts

```diff
@@ -2,6 +2,14 @@ import { DataQuery, DataSourceJsonData } from '@grafana/ui/src/types';
 
 export interface PromQuery extends DataQuery {
   expr: string;
+  container?: 'explore' | 'panel';
```

Comment:
Not sure about the naming of `container`. Naming is always hard :smile: Feels more like something related to context, either you're in Explore or in Dashboard?

Reply:
I like context, will use that instead. Thanks!

Reply:
On second thought feels like context should be a property one level up and something that Explore provides when calling `datasource.query`.